src/routes/group.route.ts

```diff
@@ -0,0 +1,27 @@
+import { Router } from 'express';
+import GroupsController from '@controllers/groupe.controller';
+import { CreateGroupDto } from '@dtos/group.dto';
+import { Routes } from '@interfaces/routes.interface';
+import validationMiddleware from '@middlewares/validation.middleware';
+
+class GroupsRoute implements Routes {
+  public path = '/groups';
+  public router = Router();
+  public groupsController = new GroupsController();
+
+  constructor() {
+    this.initializeRoutes();
+  }
+
+  private initializeRoutes() {
+    this.router.get(`${this.path}`, this.groupsController.getGroups);
+    this.router.get(`${this.path}/:id(\\d+)`, this.groupsController.getGroupById);
+    this.router.post(`${this.path}`, validationMiddleware(CreateGroupDto, 'body'), this.groupsController.createGroup);
+    this.router.put(`${this.path}/:id(\\d+)`, validationMiddleware(CreateGroupDto, 'body', true), this.groupsController.updateGroup);
+    this.router.delete(`${this.path}/:id(\\d+)`, this.groupsController.deleteGroup);
+    this.router.put(`${this.path}/join/:id(\\d+)`, this.groupsController.joinGroup);
+    this.router.put(`${this.path}/leave/:id(\\d+)`, this.groupsController.leaveGroup);
+  }
+}
+
+export default GroupsRoute;
```

Comment:
1. Group your routes based on their HTTP Method: As your API grows, the number of methods you use can increase and make the code harder to read. Consider grouping your routes according to their HTTP methods like GET, POST, PUT, DELETE, etc. You can place these groupings in their method blocks for better organization of code.

2. Separate static path strings and use meaningful names: Instead of putting entire URLs (including static parts) inside http method calls like get or put, create constants with meaningful names before your method calls and use them in the calls.

3. Error Handling: It seems like your code isn't currently handling exceptions. If an exception occurs within your promise chain (for example an error during the getGroups call in GroupsController,) it will not be handled gracefully and could crash your application. Consider using a global error handler or wrapping your route handlers with a catch-all error handler to ensure that all errors are handled.

4. Use Asynchronous Functions: Depending on what your controller methods do, they might benefit from being declared async. This could potentially also help with unhandled promise rejections if you're not careful.

5. Explicit Function Return Types: While TypeScript does a good job of inferencing types, it's a good practice to define return types for all public API methods. If the function is not expected to return anything, use void. This ensures that changes that might cause a function to inadvertently return a value will be caught.

6. Route Validation: You've started to validate your data using a middleware. That's good, but consider using a separate validation schema for PUT requests (like UpdateGroupDto), since generally, for an update not all fields are required unlike the creation of a new object.

This is example of what we can do with all this suggestions:
```
class GroupsRoute implements Routes {
  public path = '/groups';
  public router = Router();
  public groupsController = new GroupsController();
  private idPath = `${this.path}/:id(\\d+)`;
  private joinPath = `${this.path}/join/:id(\\d+)`;
  private leavePath = `${this.path}/leave/:id(\\d+)`;

  constructor() {
    this.initializeRoutes();
  }

  private handleError = (fn) => (req, res, next) => {
    Promise.resolve(fn(req, res, next)).catch(next);
  };

  private initializeRoutes() {
    this.router.get(this.path, this.handleError(async (req, res, next) => {
      const groups = await this.groupsController.getGroups(req, res);
      res.json(groups);
    }));

    this.router.post(this.path, validationMiddleware(CreateGroupDto, 'body'), this.handleError(async (req, res, next) => {
      const group = await this.groupsController.createGroup(req, res);
      res.json(group);
    }));

    this.router.get(this.idPath, this.handleError(async (req, res, next) => {
      const group = await this.groupsController.getGroupById(req, res);
      res.json(group);
    }));

    this.router.put(this.idPath, validationMiddleware(CreateGroupDto, 'body', true), this.handleError(async (req, res, next) => {
      const updatedGroup = await this.groupsController.updateGroup(req, res);
      res.json(updatedGroup);
    }));

    this.router.delete(this.idPath, this.handleError(async (req, res, next) => {
      await this.groupsController.deleteGroup(req, res);
      res.json({message: "Group successfully deleted"});
    }));

    this.router.put(this.joinPath, this.handleError(async (req, res, next) => {
      await this.groupsController.joinGroup(req, res);
      res.json({message: "Successfully joined group"});
    }));

    this.router.put(this.leavePath, this.handleError(async (req, res, next) => {
      await this.groupsController.leaveGroup(req, res);
      res.json({message: "Successfully left group"});
    }));
  }
}

export default GroupsRoute;
```